policy/reviewer/reviewer.go

```diff
@@ -124,31 +127,94 @@ func FindRandomRequesters(ctx context.Context, prctx pull.Context, result common
-			for user, perm := range allCollaboratorPermissions {
-				if perm == common.GithubAdminPermission {
-					repoAdmins = append(repoAdmins, user)
-				}
-			}
-			shoveIntoMap(allUsers, repoAdmins)
-		}
-
 		if child.ReviewRequestRule.WriteCollaborators {
 			var repoCollaborators []string
-			for user, perm := range allCollaboratorPermissions {
+			for user, perm := range allCollaboratorAndPermissions {
 				if perm == common.GithubWritePermission {
 					repoCollaborators = append(repoCollaborators, user)
 				}
 			}
 			shoveIntoMap(allUsers, repoCollaborators)
 		}
 
+		// When admins are selected for review, only collect the desired set of admins instead of
+		// everyone, which includes org admins
+		if !child.ReviewRequestRule.Admins {
+			// When not looking for admins, we want to check with everyone
+			allCollaboratorPermissions = allCollaboratorAndPermissions
+		} else {
+			// Determine what the scope of requested admins should be
+			switch child.ReviewRequestRule.AdminScope {
+			case common.User:
+				logger.Debug().Msg("Selecting admin users with direct collaboration rights")
+				directCollaborators, err := prctx.DirectRepositoryCollaborators()
+				if err != nil {
+					logger.Error().Err(err).Msgf("Unable to get list of direct collaborators on %s", prctx.RepositoryName())
+				}
+
+				var repoAdmins []string
+				for user, perm := range directCollaborators {
+					if perm == common.GithubAdminPermission {
+						repoAdmins = append(repoAdmins, user)
+					}
+				}
+				shoveIntoMap(allUsers, repoAdmins)
+				for _, a := range repoAdmins {
+					allCollaboratorPermissions[a] = common.GithubAdminPermission
+				}
+
+				break
+			case common.Team:
+				// Only request review for admins that are added as a team
+				// Resolve all admin teams on the repo, and resolve their user membership
+				logger.Debug().Msg("Selecting admin users from teams")
+				teams, err := prctx.Teams()
+				if err != nil {
+					logger.Error().Err(err).Msg("Unable to get list of teams collaborators")
+				}
+
+				var adminUsers []string
+				for team, perm := range teams {
+					if perm == common.GithubAdminPermission {
+						fullTeamName := fmt.Sprintf("%s/%s", prctx.RepositoryOwner(), team)
+						admins, err := prctx.TeamMembers(fullTeamName)
+						if err != nil {
+							logger.Error().Err(err).Msgf("Unable to get list of members for %s", team)
+						}
+						adminUsers = append(adminUsers, admins...)
+					}
+				}
+
+				shoveIntoMap(allUsers, adminUsers)
+				for _, a := range adminUsers {
+					allCollaboratorPermissions[a] = common.GithubAdminPermission
+				}
+
+				break
+			case common.Org:
+				logger.Debug().Msg("Selecting admin users from the org")
+				orgOwners, err := prctx.OrganizationOwners(prctx.RepositoryOwner())
+				if err != nil {
+					logger.Error().Err(err).Msgf("Unable to get list of org owners for %s", prctx.RepositoryOwner())
+				}
+
+				for _, o := range orgOwners {
+					allUsers[o] = struct{}{}
```

Comment:
Why do the other cases use `shoveInToMap`, but this adds the users directly in a single loop? I think the single loop version (this one) is preferable, but would rather have consistency.

Reply:
I'll remove the `shoveIntoMap` calls